File: 215. Kth Largest Element in an Array.cpp
#include"LeetCodeLibs.h"
class Solution1 {
public:
    int findKthLargest(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());


        return nums[nums.size() - k];
    }   
};

class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, less<int>> maxHeap;

        for(int& i : nums){
            maxHeap.push(i);
        }

        while(--k) maxHeap.pop();

        return maxHeap.top();
    }
};          

File: 215. Kth Largest Element in an Array.java

import java.util.Arrays;
import java.util.Collections;
import java.util.PriorityQueue;



class Solution {
    public int findKthLargest(int[] nums, int k) {
        Arrays.sort(nums);

        return nums[nums.length - k];
    }
}


class Solution2 {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>(Collections.reverseOrder());

        for(int i : nums) maxHeap.offer(i);

        while(k > 1){
            --k;
            maxHeap.poll();
        }

        return maxHeap.peek();
    }
}